### PR TITLE
Remove NativeMethods, SafeNativeMethods and UnsafeNativeMethods classes from DesignerActionPanel.cs

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- SafeNativeMethods class seems to be unused, so remove it.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2221)

### CODE REVIEW

**Comment by gpetrou** [comment]
@RussKie is it OK to use InternalsVisibleTo from Forms to Design to fix the build error?

**Comment by JeremyKuhne** [comment]
> is it OK to use InternalsVisibleTo from Forms to Design to fix the build error?

Looking for a good option to deal with `IHandle`...

**Comment by RussKie** [comment]
@JeremyKuhne assigning to you for now

**Comment by gpetrou** [comment]
I assume another option would be to move IHandle to Common code on a separate PR first?

**Comment by RussKie** [comment]
@gpetrou https://github.com/dotnet/winforms/pull/2348 should give you a way forward with this work. Please rebase.
Thank you

**Comment by gpetrou** [comment]
@RussKie This is ready now.
Thanks @JeremyKuhne.

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.cs [hunk 1]
```diff
@@ -2449,12 +2449,12 @@ public void ShowDropDown(Control parent)
                 {
                     try
                     {
-                        UnsafeNativeMethods.SetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_HWNDPARENT, new HandleRef(parent, parent.Handle));
+                        User32.SetWindowLong(Handle, User32.GWL.HWNDPARENT, parent.Handle);
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.cs]: In these ones we can't remove the `this` reference. You can add `IHandle` to `DesignerActionPanel` and pass `this`.

Technically `this` can be collected as soon as it isn't referenced any more. We could evaluate self-reference here, but it's prone to accidentally break with further code changes.

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.cs [hunk 2]
```diff
@@ -2464,7 +2464,7 @@ public void ShowDropDown(Control parent)
                     finally
                     {
 
-                        UnsafeNativeMethods.SetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_HWNDPARENT, new HandleRef(null, IntPtr.Zero));
+                        User32.SetWindowLong(Handle, User32.GWL.HWNDPARENT, IntPtr.Zero);
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.cs]: Same here, `this` owns `this.Handle`.

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.cs (hunk 1 above)
Comment (gpetrou) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.cs]: Sorry, wasn't sure if you wanted me to inherit from IHandle or not.
I have added an overload that uses HandleRef instead.

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.cs [hunk 3]
```diff
@@ -2449,12 +2449,12 @@ public void ShowDropDown(Control parent)
                 {
                     try
                     {
-                        UnsafeNativeMethods.SetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_HWNDPARENT, new HandleRef(parent, parent.Handle));
+                        User32.SetWindowLong(new HandleRef(this, Handle), User32.GWL.HWNDPARENT, parent.Handle);
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.cs]:
```suggestion
                        User32.SetWindowLong(this, User32.GWL.HWNDPARENT, parent.Handle);
```

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.cs [hunk 4]
```diff
@@ -2464,7 +2464,7 @@ public void ShowDropDown(Control parent)
                     finally
                     {
 
-                        UnsafeNativeMethods.SetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_HWNDPARENT, new HandleRef(null, IntPtr.Zero));
+                        User32.SetWindowLong(new HandleRef(this, Handle), User32.GWL.HWNDPARENT, IntPtr.Zero);
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.cs]:
```suggestion
                        User32.SetWindowLong(this, User32.GWL.HWNDPARENT, IntPtr.Zero);
```

File: src/Common/src/Interop/User32/Interop.SetWindowLong.cs [hunk 5]
```diff
@@ -33,6 +33,13 @@ public static IntPtr SetWindowLong(IHandle hWnd, GWL nIndex, IntPtr dwNewLong)
             return result;
         }
 
+        public static IntPtr SetWindowLong(HandleRef hWnd, GWL nIndex, IntPtr dwNewLong)
+        {
+            IntPtr result = SetWindowLong(hWnd.Handle, nIndex, dwNewLong);
+            GC.KeepAlive(hWnd.Handle);
+            return result;
+        }
```
Comment (RussKie) [inline src/Common/src/Interop/User32/Interop.SetWindowLong.cs]: I don't believe this is required

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.cs [hunk 6]
```diff
@@ -2449,12 +2449,12 @@ public void ShowDropDown(Control parent)
                 {
                     try
                     {
-                        UnsafeNativeMethods.SetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_HWNDPARENT, new HandleRef(parent, parent.Handle));
+                        User32.SetWindowLong(this, User32.GWL.HWNDPARENT, parent.Handle);
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.cs]:
```suggestion
                        User32.SetWindowLong(this, User32.GWL.HWNDPARENT, parent);
```

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.cs [hunk 7]
```diff
@@ -2449,12 +2449,12 @@ public void ShowDropDown(Control parent)
...
                         // Lifted directly from Form.ShowDialog()...
                         IntPtr hWndCapture = User32.GetCapture();
                         if (hWndCapture != IntPtr.Zero)
                         {
-                            UnsafeNativeMethods.SendMessage(new HandleRef(null, hWndCapture), WindowMessages.WM_CANCELMODE, 0, 0);
+                            User32.SendMessageW(hWndCapture, User32.WindowMessage.WM_CANCELMODE, IntPtr.Zero, IntPtr.Zero);
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.cs]:
```suggestion
                            User32.SendMessageW(hWndCapture, User32.WindowMessage.WM_CANCELMODE);
```

**Comment by RussKie** [comment]
There are multiple build failures, something is broken

**Comment by RussKie** [comment]
Thank you

**Pull request merged**